nats-concepts/jetstream/source_and_mirror.md

```diff
@@ -0,0 +1,69 @@
+
+The stream using a source or mirror configuration can have its own retention policy, replication, and storage type.
+
+{% hint style="info" %}
+* Changes to to the stream using source or mirror,e.g. deleting messages or publishing, do not reflect back on the origin stream from which the data was received.
+* Deletes in the origin stream are NOT replicated through a `source` or `mirror` agreement.
+{% endhint %}
+
+{% hint style="info" %}
+`Sources` is a generalization of the `Mirror` and allows for sourcing data from one or more streams concurrently. 
+If you require the target stream to act as a read-only replica:
+* Configure the stream without listen subjects **or**
+* Temporarily disable the listen subjects through client authorizations. 
+{% endhint %}
+
+## Source specific
+A stream defining `Sources` is a generalized replication mechanism and allows for sourcing data from **one or more streams** concurrently as well as allowing direct write/publish by clients. Essentially the source streams and client writes are aggregated into a single interleaved stream.
+Subject transformation and filtering allow for powerful data distribution architectures.
+
+{% hint style="info" %}
+Sourcing messages does not retain sequence numbers. But it retain the in stream sequence of messages . Between streams sourced to the same target, the sequence of messages is undefined.
+{% endhint %}
+
+## Mirror specific
+A mirror can source its messages from **exactly one stream** and a clients can not directly write to the  mirror. Although messages cannot be published to a mirror directly by clients, messages can be deleted on-demand (beyond the retention policy), and consumers have all capabilities available on regular streams.
+
+{% hint style="info" %}
+* Mirrored messages retains the sequence numbers and timestamps of the origin stream. 
+* Mirrors can be used for for (geographic) load distribution with the `MirrorDirect` stream attribute. See: [https://docs.nats.io/nats-concepts/jetstream/streams#configuration](https://docs.nats.io/nats-concepts/jetstream/streams#configuration)
+
+{% endhint %}
+
+## Expected behavior in edge conditions
+
+* Source and mirror contracts are designed with one-way (geographic) data replication in mind. Neither configuration provides full synchronization between streams, which would include deletes or replication of other stream attributes.
+
+* The content of the stream from which a source or mirror is drawn needs to be reasonable stable. Quickly deleting messages after publishing them may result in inconsistent replication due to the asynchronous nature of the replication process.
+
+* Sources and Mirror try to be be efficient in replicating messages and are lenient towards the source/mirror origin temporarily not being reachable, e.g. when using leaf nodes, which are connected intermittently. For sake of efficiency the recovery interval in case of a disconnect is 10-20s.
```

Comment:
Technically it's not temporary, you could have the sourcing stream not being able to see the stream being sourced for one year and it would still recover the missed messages, the limit is more the limits of the retention of messages on the stream being sourced/mirrored (ie if the disconnect is 1 year and the stream being sourced has a max age of 1 month, after one year the most it will be able to recover is the last month).

It is also expected that when adding a source to a stream, it can take up to a few seconds for the sourcing to start (even if the stream being sourced is up and reachable).

Reply:
Explicitley descibed intermittently connected leaf nodes as a pattern